### DISCUSSION TITLE
displayer rerender only when holding left click on displayer.

### DISCUSSION THREAD

**Discussion by MeyerBuaharon** [isl-org/Open3D#4605] [Q&A]
I made a code in cpp using open3d, and I tried to make it so I run a number of PLYs and show them one after the other.

`

#include <unistd.h>

#include <string>
#include <thread>

#include "open3d/Open3D.h"
#include "open3d/geometry/TriangleMesh.h"
#include "open3d/io/TriangleMeshIO.h"
#include "open3d/utility/DataManager.h"
#include "open3d/visualization/visualizer/GuiVisualizer.h"

using namespace open3d;
using namespace open3d::visualization;
using namespace std;

std::shared_ptr<geometry::PointCloud> pointcloud_ptr_A(new geometry::PointCloud);

void LoadPLY(geometry::PointCloud& pointcloud_ptr,
             char pointcloud_type,
             int index) {
	if (pointcloud_type == 'A') {
        io::ReadPointCloud("../SplitCube/" + std::to_string(index) + "A.ply",
                           pointcloud_ptr);
    }
	// does something else...
}

void StartPointCloud(std::shared_ptr<gui::SceneWidget> widget, std::shared_ptr<open3d::visualization::visualizer::O3DVisualizer> window, int index)
{

    LoadPLY(*(pointcloud_ptr_A.get()), 'B', index);
 
	window->GetScene()->ClearGeometry();

    window->GetScene()->AddGeometry("pcdA", pointcloud_ptr_A.get(),
                                    rendering::MaterialRecord());

}
void run(std::shared_ptr<open3d::visualization::visualizer::O3DVisualizer> window, std::shared_ptr<gui::SceneWidget> widget)
{
    int index = 0;
    while (true) 
	{
        std::cout << "Thread Is Running:" << std::endl;
        
        index++;
        if (index == 100)
		{
            index = 1;
        }

        thread t1([index, widget, window]() {
            StartPointCloud(widget, window, index);
        });
		t1.detach();

		this_thread::sleep_for(chrono::milliseconds(40));
    }
}

int main(int argc, char* argv[]) {
    int width = 1024;
    int height = 768;
    std::string window_name = "Open3D";
    Eigen::Vector3f center = {0.0, 0.0, 0.0};
    Eigen::Vector3f eye = {125.0, 125.0, 125.0};
    Eigen::Vector3f up = {0.0, 0.0, 1.0};

    auto& app = gui::Application::GetInstance();
    app.Initialize();

    auto window = std::make_shared<open3d::visualization::visualizer::O3DVisualizer>(window_name, width, height);
    app.AddWindow(window);

    auto widget = std::make_shared<gui::SceneWidget>();

    widget->SetScene(std::make_shared<rendering::Open3DScene>(window->GetRenderer()));
    window->AddChild(widget);

    auto mat = rendering::MaterialRecord();
    mat.shader = "defaultLit";
    window->GetScene()->GetCamera()->LookAt(center, eye, up);

    thread t([window, widget]() { run(window, widget); });
    t.detach();

    app.Run();

    return 0;
}

`

Can I get any pointers on what I did wrong here?
